src/libstd/ffi/c_str.rs

```diff
@@ -1305,6 +1305,36 @@ impl AsRef<CStr> for CString {
     }
 }
 
+#[unstable(feature = "slice_concat_ext",
+           reason = "trait should not have to exist",
+           issue = "27747")]
+impl<S: Borrow<CStr>> SliceConcatExt<OsStr> for [S] {
+    type Output = CString;
+
+    fn concat(&self) -> CString {
+        self.join(unsafe { CStr::from_bytes_with_nul_unchecked(&[0]) })
```

Comment:
This is unsafe right now, and can be refactored to `CStr::from_bytes_with_nul(&[0]).unwrap()`

Reply:
However, it is always going to be `Ok`